// @aws-sdk/client-timestream-write/dist-es/commands/WriteRecordsCommand.js
import { getEndpointPlugin } from "@aws-sdk/middleware-endpoint";
import { getEndpointDiscoveryPlugin } from "@aws-sdk/middleware-endpoint-discovery";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { WriteRecordsRequestFilterSensitiveLog, WriteRecordsResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_0WriteRecordsCommand, serializeAws_json1_0WriteRecordsCommand, } from "../protocols/Aws_json1_0";
export class WriteRecordsCommand extends $Command {
    constructor(input) {
        super();
        this.input = input;
    }
    static getEndpointParameterInstructions() {
        return {
            UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
            Endpoint: { type: "builtInParams", name: "endpoint" },
            Region: { type: "builtInParams", name: "region" },
            UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
        };
    }
    resolveMiddleware(clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        this.middlewareStack.use(getEndpointPlugin(configuration, WriteRecordsCommand.getEndpointParameterInstructions()));
        this.middlewareStack.use(getEndpointDiscoveryPlugin(configuration, { clientStack, options, isDiscoveredEndpointRequired: true }));
        const stack = clientStack.concat(this.middlewareStack);
        const { logger } = configuration;
        const clientName = "TimestreamWriteClient";
        const commandName = "WriteRecordsCommand";
        const handlerExecutionContext = {
            logger,
            clientName,
            commandName,
            inputFilterSensitiveLog: WriteRecordsRequestFilterSensitiveLog,
            outputFilterSensitiveLog: WriteRecordsResponseFilterSensitiveLog,
        };
        const { requestHandler } = configuration;
        return stack.resolve((request) => requestHandler.handle(request.request, options || {}), handlerExecutionContext);
    }
    serialize(input, context) {
        return serializeAws_json1_0WriteRecordsCommand(input, context);
    }
    deserialize(output, context) {
        return deserializeAws_json1_0WriteRecordsCommand(output, context);
    }
}
