// @aws-sdk/client-timestream-query/dist-es/models/models_0.js
import { SENSITIVE_STRING } from "@aws-sdk/smithy-client";
import { TimestreamQueryServiceException as __BaseException } from "./TimestreamQueryServiceException";
export class AccessDeniedException extends __BaseException {
    constructor(opts) {
        super({
            name: "AccessDeniedException",
            $fault: "client",
            ...opts,
        });
        this.name = "AccessDeniedException";
        this.$fault = "client";
        Object.setPrototypeOf(this, AccessDeniedException.prototype);
        this.Message = opts.Message;
    }
}
export class InternalServerException extends __BaseException {
    constructor(opts) {
        super({
            name: "InternalServerException",
            $fault: "server",
            ...opts,
        });
        this.name = "InternalServerException";
        this.$fault = "server";
        Object.setPrototypeOf(this, InternalServerException.prototype);
        this.Message = opts.Message;
    }
}
export class InvalidEndpointException extends __BaseException {
    constructor(opts) {
        super({
            name: "InvalidEndpointException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidEndpointException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidEndpointException.prototype);
        this.Message = opts.Message;
    }
}
export class ThrottlingException extends __BaseException {
    constructor(opts) {
        super({
            name: "ThrottlingException",
            $fault: "client",
            ...opts,
        });
        this.name = "ThrottlingException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ThrottlingException.prototype);
        this.Message = opts.Message;
    }
}
export class ValidationException extends __BaseException {
    constructor(opts) {
        super({
            name: "ValidationException",
            $fault: "client",
            ...opts,
        });
        this.name = "ValidationException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ValidationException.prototype);
        this.Message = opts.Message;
    }
}
export var ScalarType;
(function (ScalarType) {
    ScalarType["BIGINT"] = "BIGINT";
    ScalarType["BOOLEAN"] = "BOOLEAN";
    ScalarType["DATE"] = "DATE";
    ScalarType["DOUBLE"] = "DOUBLE";
    ScalarType["INTEGER"] = "INTEGER";
    ScalarType["INTERVAL_DAY_TO_SECOND"] = "INTERVAL_DAY_TO_SECOND";
    ScalarType["INTERVAL_YEAR_TO_MONTH"] = "INTERVAL_YEAR_TO_MONTH";
    ScalarType["TIME"] = "TIME";
    ScalarType["TIMESTAMP"] = "TIMESTAMP";
    ScalarType["UNKNOWN"] = "UNKNOWN";
    ScalarType["VARCHAR"] = "VARCHAR";
})(ScalarType || (ScalarType = {}));
export class ConflictException extends __BaseException {
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        this.name = "ConflictException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ConflictException.prototype);
        this.Message = opts.Message;
    }
}
export var S3EncryptionOption;
(function (S3EncryptionOption) {
    S3EncryptionOption["SSE_KMS"] = "SSE_KMS";
    S3EncryptionOption["SSE_S3"] = "SSE_S3";
})(S3EncryptionOption || (S3EncryptionOption = {}));
export var DimensionValueType;
(function (DimensionValueType) {
    DimensionValueType["VARCHAR"] = "VARCHAR";
})(DimensionValueType || (DimensionValueType = {}));
export var MeasureValueType;
(function (MeasureValueType) {
    MeasureValueType["BIGINT"] = "BIGINT";
    MeasureValueType["BOOLEAN"] = "BOOLEAN";
    MeasureValueType["DOUBLE"] = "DOUBLE";
    MeasureValueType["MULTI"] = "MULTI";
    MeasureValueType["VARCHAR"] = "VARCHAR";
})(MeasureValueType || (MeasureValueType = {}));
export var ScalarMeasureValueType;
(function (ScalarMeasureValueType) {
    ScalarMeasureValueType["BIGINT"] = "BIGINT";
    ScalarMeasureValueType["BOOLEAN"] = "BOOLEAN";
    ScalarMeasureValueType["DOUBLE"] = "DOUBLE";
    ScalarMeasureValueType["TIMESTAMP"] = "TIMESTAMP";
    ScalarMeasureValueType["VARCHAR"] = "VARCHAR";
})(ScalarMeasureValueType || (ScalarMeasureValueType = {}));
export class ServiceQuotaExceededException extends __BaseException {
    constructor(opts) {
        super({
            name: "ServiceQuotaExceededException",
            $fault: "client",
            ...opts,
        });
        this.name = "ServiceQuotaExceededException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ServiceQuotaExceededException.prototype);
        this.Message = opts.Message;
    }
}
export class ResourceNotFoundException extends __BaseException {
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        this.name = "ResourceNotFoundException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
        this.Message = opts.Message;
        this.ScheduledQueryArn = opts.ScheduledQueryArn;
    }
}
export var ScheduledQueryRunStatus;
(function (ScheduledQueryRunStatus) {
    ScheduledQueryRunStatus["AUTO_TRIGGER_FAILURE"] = "AUTO_TRIGGER_FAILURE";
    ScheduledQueryRunStatus["AUTO_TRIGGER_SUCCESS"] = "AUTO_TRIGGER_SUCCESS";
    ScheduledQueryRunStatus["MANUAL_TRIGGER_FAILURE"] = "MANUAL_TRIGGER_FAILURE";
    ScheduledQueryRunStatus["MANUAL_TRIGGER_SUCCESS"] = "MANUAL_TRIGGER_SUCCESS";
})(ScheduledQueryRunStatus || (ScheduledQueryRunStatus = {}));
export var ScheduledQueryState;
(function (ScheduledQueryState) {
    ScheduledQueryState["DISABLED"] = "DISABLED";
    ScheduledQueryState["ENABLED"] = "ENABLED";
})(ScheduledQueryState || (ScheduledQueryState = {}));
export class QueryExecutionException extends __BaseException {
    constructor(opts) {
        super({
            name: "QueryExecutionException",
            $fault: "client",
            ...opts,
        });
        this.name = "QueryExecutionException";
        this.$fault = "client";
        Object.setPrototypeOf(this, QueryExecutionException.prototype);
        this.Message = opts.Message;
    }
}
export const CancelQueryRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CancelQueryResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const S3ConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ErrorReportConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SnsConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const NotificationConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ScheduleConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const TagFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DimensionMappingFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const MultiMeasureAttributeMappingFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const MixedMeasureMappingFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const MultiMeasureMappingsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const TimestreamConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const TargetConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const CreateScheduledQueryRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.QueryString && { QueryString: SENSITIVE_STRING }),
    ...(obj.ClientToken && { ClientToken: SENSITIVE_STRING }),
});
export const CreateScheduledQueryResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DeleteScheduledQueryRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DescribeEndpointsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const EndpointFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DescribeEndpointsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DescribeScheduledQueryRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const S3ReportLocationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ErrorReportLocationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ExecutionStatsFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ScheduledQueryRunSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ScheduledQueryDescriptionFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.QueryString && { QueryString: SENSITIVE_STRING }),
});
export const DescribeScheduledQueryResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.ScheduledQuery && { ScheduledQuery: ScheduledQueryDescriptionFilterSensitiveLog(obj.ScheduledQuery) }),
});
export const ExecuteScheduledQueryRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.ClientToken && { ClientToken: SENSITIVE_STRING }),
});
export const ListScheduledQueriesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const TimestreamDestinationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const TargetDestinationFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ScheduledQueryFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ListScheduledQueriesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ListTagsForResourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ListTagsForResourceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const PrepareQueryRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.QueryString && { QueryString: SENSITIVE_STRING }),
});
export const QueryRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.QueryString && { QueryString: SENSITIVE_STRING }),
    ...(obj.ClientToken && { ClientToken: SENSITIVE_STRING }),
});
export const QueryStatusFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const TagResourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const TagResourceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UntagResourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UntagResourceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const UpdateScheduledQueryRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ColumnInfoFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const TypeFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const DatumFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const ParameterMappingFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const SelectColumnFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const TimeSeriesDataPointFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const RowFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const QueryResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
export const PrepareQueryResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.QueryString && { QueryString: SENSITIVE_STRING }),
});
