// @aws-sdk/client-timestream-write/dist-cjs/protocols/Aws_json1_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deserializeAws_json1_0WriteRecordsCommand = exports.deserializeAws_json1_0UpdateTableCommand = exports.deserializeAws_json1_0UpdateDatabaseCommand = exports.deserializeAws_json1_0UntagResourceCommand = exports.deserializeAws_json1_0TagResourceCommand = exports.deserializeAws_json1_0ListTagsForResourceCommand = exports.deserializeAws_json1_0ListTablesCommand = exports.deserializeAws_json1_0ListDatabasesCommand = exports.deserializeAws_json1_0DescribeTableCommand = exports.deserializeAws_json1_0DescribeEndpointsCommand = exports.deserializeAws_json1_0DescribeDatabaseCommand = exports.deserializeAws_json1_0DeleteTableCommand = exports.deserializeAws_json1_0DeleteDatabaseCommand = exports.deserializeAws_json1_0CreateTableCommand = exports.deserializeAws_json1_0CreateDatabaseCommand = exports.serializeAws_json1_0WriteRecordsCommand = exports.serializeAws_json1_0UpdateTableCommand = exports.serializeAws_json1_0UpdateDatabaseCommand = exports.serializeAws_json1_0UntagResourceCommand = exports.serializeAws_json1_0TagResourceCommand = exports.serializeAws_json1_0ListTagsForResourceCommand = exports.serializeAws_json1_0ListTablesCommand = exports.serializeAws_json1_0ListDatabasesCommand = exports.serializeAws_json1_0DescribeTableCommand = exports.serializeAws_json1_0DescribeEndpointsCommand = exports.serializeAws_json1_0DescribeDatabaseCommand = exports.serializeAws_json1_0DeleteTableCommand = exports.serializeAws_json1_0DeleteDatabaseCommand = exports.serializeAws_json1_0CreateTableCommand = exports.serializeAws_json1_0CreateDatabaseCommand = void 0;
const protocol_http_1 = require("@aws-sdk/protocol-http");
const smithy_client_1 = require("@aws-sdk/smithy-client");
const models_0_1 = require("../models/models_0");
const TimestreamWriteServiceException_1 = require("../models/TimestreamWriteServiceException");
const serializeAws_json1_0CreateDatabaseCommand = async (input, context) => {
    const headers = {
        "content-type": "application/x-amz-json-1.0",
        "x-amz-target": "Timestream_20181101.CreateDatabase",
    };
    let body;
    body = JSON.stringify(serializeAws_json1_0CreateDatabaseRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.serializeAws_json1_0CreateDatabaseCommand = serializeAws_json1_0CreateDatabaseCommand;
const serializeAws_json1_0CreateTableCommand = async (input, context) => {
    const headers = {
        "content-type": "application/x-amz-json-1.0",
        "x-amz-target": "Timestream_20181101.CreateTable",
    };
    let body;
    body = JSON.stringify(serializeAws_json1_0CreateTableRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.serializeAws_json1_0CreateTableCommand = serializeAws_json1_0CreateTableCommand;
const serializeAws_json1_0DeleteDatabaseCommand = async (input, context) => {
    const headers = {
        "content-type": "application/x-amz-json-1.0",
        "x-amz-target": "Timestream_20181101.DeleteDatabase",
    };
    let body;
    body = JSON.stringify(serializeAws_json1_0DeleteDatabaseRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.serializeAws_json1_0DeleteDatabaseCommand = serializeAws_json1_0DeleteDatabaseCommand;
const serializeAws_json1_0DeleteTableCommand = async (input, context) => {
    const headers = {
        "content-type": "application/x-amz-json-1.0",
        "x-amz-target": "Timestream_20181101.DeleteTable",
    };
    let body;
    body = JSON.stringify(serializeAws_json1_0DeleteTableRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.serializeAws_json1_0DeleteTableCommand = serializeAws_json1_0DeleteTableCommand;
const serializeAws_json1_0DescribeDatabaseCommand = async (input, context) => {
    const headers = {
        "content-type": "application/x-amz-json-1.0",
        "x-amz-target": "Timestream_20181101.DescribeDatabase",
    };
    let body;
    body = JSON.stringify(serializeAws_json1_0DescribeDatabaseRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.serializeAws_json1_0DescribeDatabaseCommand = serializeAws_json1_0DescribeDatabaseCommand;
const serializeAws_json1_0DescribeEndpointsCommand = async (input, context) => {
    const headers = {
        "content-type": "application/x-amz-json-1.0",
        "x-amz-target": "Timestream_20181101.DescribeEndpoints",
    };
    let body;
    body = JSON.stringify(serializeAws_json1_0DescribeEndpointsRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.serializeAws_json1_0DescribeEndpointsCommand = serializeAws_json1_0DescribeEndpointsCommand;
const serializeAws_json1_0DescribeTableCommand = async (input, context) => {
    const headers = {
        "content-type": "application/x-amz-json-1.0",
        "x-amz-target": "Timestream_20181101.DescribeTable",
    };
    let body;
    body = JSON.stringify(serializeAws_json1_0DescribeTableRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.serializeAws_json1_0DescribeTableCommand = serializeAws_json1_0DescribeTableCommand;
const serializeAws_json1_0ListDatabasesCommand = async (input, context) => {
    const headers = {
        "content-type": "application/x-amz-json-1.0",
        "x-amz-target": "Timestream_20181101.ListDatabases",
    };
    let body;
    body = JSON.stringify(serializeAws_json1_0ListDatabasesRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.serializeAws_json1_0ListDatabasesCommand = serializeAws_json1_0ListDatabasesCommand;
const serializeAws_json1_0ListTablesCommand = async (input, context) => {
    const headers = {
        "content-type": "application/x-amz-json-1.0",
        "x-amz-target": "Timestream_20181101.ListTables",
    };
    let body;
    body = JSON.stringify(serializeAws_json1_0ListTablesRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.serializeAws_json1_0ListTablesCommand = serializeAws_json1_0ListTablesCommand;
const serializeAws_json1_0ListTagsForResourceCommand = async (input, context) => {
    const headers = {
        "content-type": "application/x-amz-json-1.0",
        "x-amz-target": "Timestream_20181101.ListTagsForResource",
    };
    let body;
    body = JSON.stringify(serializeAws_json1_0ListTagsForResourceRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.serializeAws_json1_0ListTagsForResourceCommand = serializeAws_json1_0ListTagsForResourceCommand;
const serializeAws_json1_0TagResourceCommand = async (input, context) => {
    const headers = {
        "content-type": "application/x-amz-json-1.0",
        "x-amz-target": "Timestream_20181101.TagResource",
    };
    let body;
    body = JSON.stringify(serializeAws_json1_0TagResourceRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.serializeAws_json1_0TagResourceCommand = serializeAws_json1_0TagResourceCommand;
const serializeAws_json1_0UntagResourceCommand = async (input, context) => {
    const headers = {
        "content-type": "application/x-amz-json-1.0",
        "x-amz-target": "Timestream_20181101.UntagResource",
    };
    let body;
    body = JSON.stringify(serializeAws_json1_0UntagResourceRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.serializeAws_json1_0UntagResourceCommand = serializeAws_json1_0UntagResourceCommand;
const serializeAws_json1_0UpdateDatabaseCommand = async (input, context) => {
    const headers = {
        "content-type": "application/x-amz-json-1.0",
        "x-amz-target": "Timestream_20181101.UpdateDatabase",
    };
    let body;
    body = JSON.stringify(serializeAws_json1_0UpdateDatabaseRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.serializeAws_json1_0UpdateDatabaseCommand = serializeAws_json1_0UpdateDatabaseCommand;
const serializeAws_json1_0UpdateTableCommand = async (input, context) => {
    const headers = {
        "content-type": "application/x-amz-json-1.0",
        "x-amz-target": "Timestream_20181101.UpdateTable",
    };
    let body;
    body = JSON.stringify(serializeAws_json1_0UpdateTableRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.serializeAws_json1_0UpdateTableCommand = serializeAws_json1_0UpdateTableCommand;
const serializeAws_json1_0WriteRecordsCommand = async (input, context) => {
    const headers = {
        "content-type": "application/x-amz-json-1.0",
        "x-amz-target": "Timestream_20181101.WriteRecords",
    };
    let body;
    body = JSON.stringify(serializeAws_json1_0WriteRecordsRequest(input, context));
    return buildHttpRpcRequest(context, headers, "/", undefined, body);
};
exports.serializeAws_json1_0WriteRecordsCommand = serializeAws_json1_0WriteRecordsCommand;
const deserializeAws_json1_0CreateDatabaseCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return deserializeAws_json1_0CreateDatabaseCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = deserializeAws_json1_0CreateDatabaseResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return Promise.resolve(response);
};
exports.deserializeAws_json1_0CreateDatabaseCommand = deserializeAws_json1_0CreateDatabaseCommand;
const deserializeAws_json1_0CreateDatabaseCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "AccessDeniedException":
        case "com.amazonaws.timestreamwrite#AccessDeniedException":
            throw await deserializeAws_json1_0AccessDeniedExceptionResponse(parsedOutput, context);
        case "ConflictException":
        case "com.amazonaws.timestreamwrite#ConflictException":
            throw await deserializeAws_json1_0ConflictExceptionResponse(parsedOutput, context);
        case "InternalServerException":
        case "com.amazonaws.timestreamwrite#InternalServerException":
            throw await deserializeAws_json1_0InternalServerExceptionResponse(parsedOutput, context);
        case "InvalidEndpointException":
        case "com.amazonaws.timestreamwrite#InvalidEndpointException":
            throw await deserializeAws_json1_0InvalidEndpointExceptionResponse(parsedOutput, context);
        case "ServiceQuotaExceededException":
        case "com.amazonaws.timestreamwrite#ServiceQuotaExceededException":
            throw await deserializeAws_json1_0ServiceQuotaExceededExceptionResponse(parsedOutput, context);
        case "ThrottlingException":
        case "com.amazonaws.timestreamwrite#ThrottlingException":
            throw await deserializeAws_json1_0ThrottlingExceptionResponse(parsedOutput, context);
        case "ValidationException":
        case "com.amazonaws.timestreamwrite#ValidationException":
            throw await deserializeAws_json1_0ValidationExceptionResponse(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            (0, smithy_client_1.throwDefaultError)({
                output,
                parsedBody,
                exceptionCtor: TimestreamWriteServiceException_1.TimestreamWriteServiceException,
                errorCode,
            });
    }
};
const deserializeAws_json1_0CreateTableCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return deserializeAws_json1_0CreateTableCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = deserializeAws_json1_0CreateTableResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return Promise.resolve(response);
};
exports.deserializeAws_json1_0CreateTableCommand = deserializeAws_json1_0CreateTableCommand;
const deserializeAws_json1_0CreateTableCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "AccessDeniedException":
        case "com.amazonaws.timestreamwrite#AccessDeniedException":
            throw await deserializeAws_json1_0AccessDeniedExceptionResponse(parsedOutput, context);
        case "ConflictException":
        case "com.amazonaws.timestreamwrite#ConflictException":
            throw await deserializeAws_json1_0ConflictExceptionResponse(parsedOutput, context);
        case "InternalServerException":
        case "com.amazonaws.timestreamwrite#InternalServerException":
            throw await deserializeAws_json1_0InternalServerExceptionResponse(parsedOutput, context);
        case "InvalidEndpointException":
        case "com.amazonaws.timestreamwrite#InvalidEndpointException":
            throw await deserializeAws_json1_0InvalidEndpointExceptionResponse(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.timestreamwrite#ResourceNotFoundException":
            throw await deserializeAws_json1_0ResourceNotFoundExceptionResponse(parsedOutput, context);
        case "ServiceQuotaExceededException":
        case "com.amazonaws.timestreamwrite#ServiceQuotaExceededException":
            throw await deserializeAws_json1_0ServiceQuotaExceededExceptionResponse(parsedOutput, context);
        case "ThrottlingException":
        case "com.amazonaws.timestreamwrite#ThrottlingException":
            throw await deserializeAws_json1_0ThrottlingExceptionResponse(parsedOutput, context);
        case "ValidationException":
        case "com.amazonaws.timestreamwrite#ValidationException":
            throw await deserializeAws_json1_0ValidationExceptionResponse(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            (0, smithy_client_1.throwDefaultError)({
                output,
                parsedBody,
                exceptionCtor: TimestreamWriteServiceException_1.TimestreamWriteServiceException,
                errorCode,
            });
    }
};
const deserializeAws_json1_0DeleteDatabaseCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return deserializeAws_json1_0DeleteDatabaseCommandError(output, context);
    }
    await collectBody(output.body, context);
    const response = {
        $metadata: deserializeMetadata(output),
    };
    return Promise.resolve(response);
};
exports.deserializeAws_json1_0DeleteDatabaseCommand = deserializeAws_json1_0DeleteDatabaseCommand;
const deserializeAws_json1_0DeleteDatabaseCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "AccessDeniedException":
        case "com.amazonaws.timestreamwrite#AccessDeniedException":
            throw await deserializeAws_json1_0AccessDeniedExceptionResponse(parsedOutput, context);
        case "InternalServerException":
        case "com.amazonaws.timestreamwrite#InternalServerException":
            throw await deserializeAws_json1_0InternalServerExceptionResponse(parsedOutput, context);
        case "InvalidEndpointException":
        case "com.amazonaws.timestreamwrite#InvalidEndpointException":
            throw await deserializeAws_json1_0InvalidEndpointExceptionResponse(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.timestreamwrite#ResourceNotFoundException":
            throw await deserializeAws_json1_0ResourceNotFoundExceptionResponse(parsedOutput, context);
        case "ThrottlingException":
        case "com.amazonaws.timestreamwrite#ThrottlingException":
            throw await deserializeAws_json1_0ThrottlingExceptionResponse(parsedOutput, context);
        case "ValidationException":
        case "com.amazonaws.timestreamwrite#ValidationException":
            throw await deserializeAws_json1_0ValidationExceptionResponse(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            (0, smithy_client_1.throwDefaultError)({
                output,
                parsedBody,
                exceptionCtor: TimestreamWriteServiceException_1.TimestreamWriteServiceException,
                errorCode,
            });
    }
};
const deserializeAws_json1_0DeleteTableCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return deserializeAws_json1_0DeleteTableCommandError(output, context);
    }
    await collectBody(output.body, context);
    const response = {
        $metadata: deserializeMetadata(output),
    };
    return Promise.resolve(response);
};
exports.deserializeAws_json1_0DeleteTableCommand = deserializeAws_json1_0DeleteTableCommand;
const deserializeAws_json1_0DeleteTableCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "AccessDeniedException":
        case "com.amazonaws.timestreamwrite#AccessDeniedException":
            throw await deserializeAws_json1_0AccessDeniedExceptionResponse(parsedOutput, context);
        case "InternalServerException":
        case "com.amazonaws.timestreamwrite#InternalServerException":
            throw await deserializeAws_json1_0InternalServerExceptionResponse(parsedOutput, context);
        case "InvalidEndpointException":
        case "com.amazonaws.timestreamwrite#InvalidEndpointException":
            throw await deserializeAws_json1_0InvalidEndpointExceptionResponse(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.timestreamwrite#ResourceNotFoundException":
            throw await deserializeAws_json1_0ResourceNotFoundExceptionResponse(parsedOutput, context);
        case "ThrottlingException":
        case "com.amazonaws.timestreamwrite#ThrottlingException":
            throw await deserializeAws_json1_0ThrottlingExceptionResponse(parsedOutput, context);
        case "ValidationException":
        case "com.amazonaws.timestreamwrite#ValidationException":
            throw await deserializeAws_json1_0ValidationExceptionResponse(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            (0, smithy_client_1.throwDefaultError)({
                output,
                parsedBody,
                exceptionCtor: TimestreamWriteServiceException_1.TimestreamWriteServiceException,
                errorCode,
            });
    }
};
const deserializeAws_json1_0DescribeDatabaseCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return deserializeAws_json1_0DescribeDatabaseCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = deserializeAws_json1_0DescribeDatabaseResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return Promise.resolve(response);
};
exports.deserializeAws_json1_0DescribeDatabaseCommand = deserializeAws_json1_0DescribeDatabaseCommand;
const deserializeAws_json1_0DescribeDatabaseCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "AccessDeniedException":
        case "com.amazonaws.timestreamwrite#AccessDeniedException":
            throw await deserializeAws_json1_0AccessDeniedExceptionResponse(parsedOutput, context);
        case "InternalServerException":
        case "com.amazonaws.timestreamwrite#InternalServerException":
            throw await deserializeAws_json1_0InternalServerExceptionResponse(parsedOutput, context);
        case "InvalidEndpointException":
        case "com.amazonaws.timestreamwrite#InvalidEndpointException":
            throw await deserializeAws_json1_0InvalidEndpointExceptionResponse(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.timestreamwrite#ResourceNotFoundException":
            throw await deserializeAws_json1_0ResourceNotFoundExceptionResponse(parsedOutput, context);
        case "ThrottlingException":
        case "com.amazonaws.timestreamwrite#ThrottlingException":
            throw await deserializeAws_json1_0ThrottlingExceptionResponse(parsedOutput, context);
        case "ValidationException":
        case "com.amazonaws.timestreamwrite#ValidationException":
            throw await deserializeAws_json1_0ValidationExceptionResponse(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            (0, smithy_client_1.throwDefaultError)({
                output,
                parsedBody,
                exceptionCtor: TimestreamWriteServiceException_1.TimestreamWriteServiceException,
                errorCode,
            });
    }
};
const deserializeAws_json1_0DescribeEndpointsCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return deserializeAws_json1_0DescribeEndpointsCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = deserializeAws_json1_0DescribeEndpointsResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return Promise.resolve(response);
};
exports.deserializeAws_json1_0DescribeEndpointsCommand = deserializeAws_json1_0DescribeEndpointsCommand;
const deserializeAws_json1_0DescribeEndpointsCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InternalServerException":
        case "com.amazonaws.timestreamwrite#InternalServerException":
            throw await deserializeAws_json1_0InternalServerExceptionResponse(parsedOutput, context);
        case "ThrottlingException":
        case "com.amazonaws.timestreamwrite#ThrottlingException":
            throw await deserializeAws_json1_0ThrottlingExceptionResponse(parsedOutput, context);
        case "ValidationException":
        case "com.amazonaws.timestreamwrite#ValidationException":
            throw await deserializeAws_json1_0ValidationExceptionResponse(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            (0, smithy_client_1.throwDefaultError)({
                output,
                parsedBody,
                exceptionCtor: TimestreamWriteServiceException_1.TimestreamWriteServiceException,
                errorCode,
            });
    }
};
const deserializeAws_json1_0DescribeTableCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return deserializeAws_json1_0DescribeTableCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = deserializeAws_json1_0DescribeTableResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return Promise.resolve(response);
};
exports.deserializeAws_json1_0DescribeTableCommand = deserializeAws_json1_0DescribeTableCommand;
const deserializeAws_json1_0DescribeTableCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "AccessDeniedException":
        case "com.amazonaws.timestreamwrite#AccessDeniedException":
            throw await deserializeAws_json1_0AccessDeniedExceptionResponse(parsedOutput, context);
        case "InternalServerException":
        case "com.amazonaws.timestreamwrite#InternalServerException":
            throw await deserializeAws_json1_0InternalServerExceptionResponse(parsedOutput, context);
        case "InvalidEndpointException":
        case "com.amazonaws.timestreamwrite#InvalidEndpointException":
            throw await deserializeAws_json1_0InvalidEndpointExceptionResponse(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.timestreamwrite#ResourceNotFoundException":
            throw await deserializeAws_json1_0ResourceNotFoundExceptionResponse(parsedOutput, context);
        case "ThrottlingException":
        case "com.amazonaws.timestreamwrite#ThrottlingException":
            throw await deserializeAws_json1_0ThrottlingExceptionResponse(parsedOutput, context);
        case "ValidationException":
        case "com.amazonaws.timestreamwrite#ValidationException":
            throw await deserializeAws_json1_0ValidationExceptionResponse(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            (0, smithy_client_1.throwDefaultError)({
                output,
                parsedBody,
                exceptionCtor: TimestreamWriteServiceException_1.TimestreamWriteServiceException,
                errorCode,
            });
    }
};
const deserializeAws_json1_0ListDatabasesCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return deserializeAws_json1_0ListDatabasesCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = deserializeAws_json1_0ListDatabasesResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return Promise.resolve(response);
};
exports.deserializeAws_json1_0ListDatabasesCommand = deserializeAws_json1_0ListDatabasesCommand;
const deserializeAws_json1_0ListDatabasesCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "AccessDeniedException":
        case "com.amazonaws.timestreamwrite#AccessDeniedException":
            throw await deserializeAws_json1_0AccessDeniedExceptionResponse(parsedOutput, context);
        case "InternalServerException":
        case "com.amazonaws.timestreamwrite#InternalServerException":
            throw await deserializeAws_json1_0InternalServerExceptionResponse(parsedOutput, context);
        case "InvalidEndpointException":
        case "com.amazonaws.timestreamwrite#InvalidEndpointException":
            throw await deserializeAws_json1_0InvalidEndpointExceptionResponse(parsedOutput, context);
        case "ThrottlingException":
        case "com.amazonaws.timestreamwrite#ThrottlingException":
            throw await deserializeAws_json1_0ThrottlingExceptionResponse(parsedOutput, context);
        case "ValidationException":
        case "com.amazonaws.timestreamwrite#ValidationException":
            throw await deserializeAws_json1_0ValidationExceptionResponse(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            (0, smithy_client_1.throwDefaultError)({
                output,
                parsedBody,
                exceptionCtor: TimestreamWriteServiceException_1.TimestreamWriteServiceException,
                errorCode,
            });
    }
};
const deserializeAws_json1_0ListTablesCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return deserializeAws_json1_0ListTablesCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = deserializeAws_json1_0ListTablesResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return Promise.resolve(response);
};
exports.deserializeAws_json1_0ListTablesCommand = deserializeAws_json1_0ListTablesCommand;
const deserializeAws_json1_0ListTablesCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "AccessDeniedException":
        case "com.amazonaws.timestreamwrite#AccessDeniedException":
            throw await deserializeAws_json1_0AccessDeniedExceptionResponse(parsedOutput, context);
        case "InternalServerException":
        case "com.amazonaws.timestreamwrite#InternalServerException":
            throw await deserializeAws_json1_0InternalServerExceptionResponse(parsedOutput, context);
        case "InvalidEndpointException":
        case "com.amazonaws.timestreamwrite#InvalidEndpointException":
            throw await deserializeAws_json1_0InvalidEndpointExceptionResponse(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.timestreamwrite#ResourceNotFoundException":
            throw await deserializeAws_json1_0ResourceNotFoundExceptionResponse(parsedOutput, context);
        case "ThrottlingException":
        case "com.amazonaws.timestreamwrite#ThrottlingException":
            throw await deserializeAws_json1_0ThrottlingExceptionResponse(parsedOutput, context);
        case "ValidationException":
        case "com.amazonaws.timestreamwrite#ValidationException":
            throw await deserializeAws_json1_0ValidationExceptionResponse(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            (0, smithy_client_1.throwDefaultError)({
                output,
                parsedBody,
                exceptionCtor: TimestreamWriteServiceException_1.TimestreamWriteServiceException,
                errorCode,
            });
    }
};
const deserializeAws_json1_0ListTagsForResourceCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return deserializeAws_json1_0ListTagsForResourceCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = deserializeAws_json1_0ListTagsForResourceResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return Promise.resolve(response);
};
exports.deserializeAws_json1_0ListTagsForResourceCommand = deserializeAws_json1_0ListTagsForResourceCommand;
const deserializeAws_json1_0ListTagsForResourceCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InvalidEndpointException":
        case "com.amazonaws.timestreamwrite#InvalidEndpointException":
            throw await deserializeAws_json1_0InvalidEndpointExceptionResponse(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.timestreamwrite#ResourceNotFoundException":
            throw await deserializeAws_json1_0ResourceNotFoundExceptionResponse(parsedOutput, context);
        case "ThrottlingException":
        case "com.amazonaws.timestreamwrite#ThrottlingException":
            throw await deserializeAws_json1_0ThrottlingExceptionResponse(parsedOutput, context);
        case "ValidationException":
        case "com.amazonaws.timestreamwrite#ValidationException":
            throw await deserializeAws_json1_0ValidationExceptionResponse(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            (0, smithy_client_1.throwDefaultError)({
                output,
                parsedBody,
                exceptionCtor: TimestreamWriteServiceException_1.TimestreamWriteServiceException,
                errorCode,
            });
    }
};
const deserializeAws_json1_0TagResourceCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return deserializeAws_json1_0TagResourceCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = deserializeAws_json1_0TagResourceResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return Promise.resolve(response);
};
exports.deserializeAws_json1_0TagResourceCommand = deserializeAws_json1_0TagResourceCommand;
const deserializeAws_json1_0TagResourceCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InvalidEndpointException":
        case "com.amazonaws.timestreamwrite#InvalidEndpointException":
            throw await deserializeAws_json1_0InvalidEndpointExceptionResponse(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.timestreamwrite#ResourceNotFoundException":
            throw await deserializeAws_json1_0ResourceNotFoundExceptionResponse(parsedOutput, context);
        case "ServiceQuotaExceededException":
        case "com.amazonaws.timestreamwrite#ServiceQuotaExceededException":
            throw await deserializeAws_json1_0ServiceQuotaExceededExceptionResponse(parsedOutput, context);
        case "ThrottlingException":
        case "com.amazonaws.timestreamwrite#ThrottlingException":
            throw await deserializeAws_json1_0ThrottlingExceptionResponse(parsedOutput, context);
        case "ValidationException":
        case "com.amazonaws.timestreamwrite#ValidationException":
            throw await deserializeAws_json1_0ValidationExceptionResponse(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            (0, smithy_client_1.throwDefaultError)({
                output,
                parsedBody,
                exceptionCtor: TimestreamWriteServiceException_1.TimestreamWriteServiceException,
                errorCode,
            });
    }
};
const deserializeAws_json1_0UntagResourceCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return deserializeAws_json1_0UntagResourceCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = deserializeAws_json1_0UntagResourceResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return Promise.resolve(response);
};
exports.deserializeAws_json1_0UntagResourceCommand = deserializeAws_json1_0UntagResourceCommand;
const deserializeAws_json1_0UntagResourceCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "InvalidEndpointException":
        case "com.amazonaws.timestreamwrite#InvalidEndpointException":
            throw await deserializeAws_json1_0InvalidEndpointExceptionResponse(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.timestreamwrite#ResourceNotFoundException":
            throw await deserializeAws_json1_0ResourceNotFoundExceptionResponse(parsedOutput, context);
        case "ServiceQuotaExceededException":
        case "com.amazonaws.timestreamwrite#ServiceQuotaExceededException":
            throw await deserializeAws_json1_0ServiceQuotaExceededExceptionResponse(parsedOutput, context);
        case "ThrottlingException":
        case "com.amazonaws.timestreamwrite#ThrottlingException":
            throw await deserializeAws_json1_0ThrottlingExceptionResponse(parsedOutput, context);
        case "ValidationException":
        case "com.amazonaws.timestreamwrite#ValidationException":
            throw await deserializeAws_json1_0ValidationExceptionResponse(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            (0, smithy_client_1.throwDefaultError)({
                output,
                parsedBody,
                exceptionCtor: TimestreamWriteServiceException_1.TimestreamWriteServiceException,
                errorCode,
            });
    }
};
const deserializeAws_json1_0UpdateDatabaseCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return deserializeAws_json1_0UpdateDatabaseCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = deserializeAws_json1_0UpdateDatabaseResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return Promise.resolve(response);
};
exports.deserializeAws_json1_0UpdateDatabaseCommand = deserializeAws_json1_0UpdateDatabaseCommand;
const deserializeAws_json1_0UpdateDatabaseCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "AccessDeniedException":
        case "com.amazonaws.timestreamwrite#AccessDeniedException":
            throw await deserializeAws_json1_0AccessDeniedExceptionResponse(parsedOutput, context);
        case "InternalServerException":
        case "com.amazonaws.timestreamwrite#InternalServerException":
            throw await deserializeAws_json1_0InternalServerExceptionResponse(parsedOutput, context);
        case "InvalidEndpointException":
        case "com.amazonaws.timestreamwrite#InvalidEndpointException":
            throw await deserializeAws_json1_0InvalidEndpointExceptionResponse(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.timestreamwrite#ResourceNotFoundException":
            throw await deserializeAws_json1_0ResourceNotFoundExceptionResponse(parsedOutput, context);
        case "ServiceQuotaExceededException":
        case "com.amazonaws.timestreamwrite#ServiceQuotaExceededException":
            throw await deserializeAws_json1_0ServiceQuotaExceededExceptionResponse(parsedOutput, context);
        case "ThrottlingException":
        case "com.amazonaws.timestreamwrite#ThrottlingException":
            throw await deserializeAws_json1_0ThrottlingExceptionResponse(parsedOutput, context);
        case "ValidationException":
        case "com.amazonaws.timestreamwrite#ValidationException":
            throw await deserializeAws_json1_0ValidationExceptionResponse(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            (0, smithy_client_1.throwDefaultError)({
                output,
                parsedBody,
                exceptionCtor: TimestreamWriteServiceException_1.TimestreamWriteServiceException,
                errorCode,
            });
    }
};
const deserializeAws_json1_0UpdateTableCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return deserializeAws_json1_0UpdateTableCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = deserializeAws_json1_0UpdateTableResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return Promise.resolve(response);
};
exports.deserializeAws_json1_0UpdateTableCommand = deserializeAws_json1_0UpdateTableCommand;
const deserializeAws_json1_0UpdateTableCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "AccessDeniedException":
        case "com.amazonaws.timestreamwrite#AccessDeniedException":
            throw await deserializeAws_json1_0AccessDeniedExceptionResponse(parsedOutput, context);
        case "InternalServerException":
        case "com.amazonaws.timestreamwrite#InternalServerException":
            throw await deserializeAws_json1_0InternalServerExceptionResponse(parsedOutput, context);
        case "InvalidEndpointException":
        case "com.amazonaws.timestreamwrite#InvalidEndpointException":
            throw await deserializeAws_json1_0InvalidEndpointExceptionResponse(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.timestreamwrite#ResourceNotFoundException":
            throw await deserializeAws_json1_0ResourceNotFoundExceptionResponse(parsedOutput, context);
        case "ThrottlingException":
        case "com.amazonaws.timestreamwrite#ThrottlingException":
            throw await deserializeAws_json1_0ThrottlingExceptionResponse(parsedOutput, context);
        case "ValidationException":
        case "com.amazonaws.timestreamwrite#ValidationException":
            throw await deserializeAws_json1_0ValidationExceptionResponse(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            (0, smithy_client_1.throwDefaultError)({
                output,
                parsedBody,
                exceptionCtor: TimestreamWriteServiceException_1.TimestreamWriteServiceException,
                errorCode,
            });
    }
};
const deserializeAws_json1_0WriteRecordsCommand = async (output, context) => {
    if (output.statusCode >= 300) {
        return deserializeAws_json1_0WriteRecordsCommandError(output, context);
    }
    const data = await parseBody(output.body, context);
    let contents = {};
    contents = deserializeAws_json1_0WriteRecordsResponse(data, context);
    const response = {
        $metadata: deserializeMetadata(output),
        ...contents,
    };
    return Promise.resolve(response);
};
exports.deserializeAws_json1_0WriteRecordsCommand = deserializeAws_json1_0WriteRecordsCommand;
const deserializeAws_json1_0WriteRecordsCommandError = async (output, context) => {
    const parsedOutput = {
        ...output,
        body: await parseErrorBody(output.body, context),
    };
    const errorCode = loadRestJsonErrorCode(output, parsedOutput.body);
    switch (errorCode) {
        case "AccessDeniedException":
        case "com.amazonaws.timestreamwrite#AccessDeniedException":
            throw await deserializeAws_json1_0AccessDeniedExceptionResponse(parsedOutput, context);
        case "InternalServerException":
        case "com.amazonaws.timestreamwrite#InternalServerException":
            throw await deserializeAws_json1_0InternalServerExceptionResponse(parsedOutput, context);
        case "InvalidEndpointException":
        case "com.amazonaws.timestreamwrite#InvalidEndpointException":
            throw await deserializeAws_json1_0InvalidEndpointExceptionResponse(parsedOutput, context);
        case "RejectedRecordsException":
        case "com.amazonaws.timestreamwrite#RejectedRecordsException":
            throw await deserializeAws_json1_0RejectedRecordsExceptionResponse(parsedOutput, context);
        case "ResourceNotFoundException":
        case "com.amazonaws.timestreamwrite#ResourceNotFoundException":
            throw await deserializeAws_json1_0ResourceNotFoundExceptionResponse(parsedOutput, context);
        case "ThrottlingException":
        case "com.amazonaws.timestreamwrite#ThrottlingException":
            throw await deserializeAws_json1_0ThrottlingExceptionResponse(parsedOutput, context);
        case "ValidationException":
        case "com.amazonaws.timestreamwrite#ValidationException":
            throw await deserializeAws_json1_0ValidationExceptionResponse(parsedOutput, context);
        default:
            const parsedBody = parsedOutput.body;
            (0, smithy_client_1.throwDefaultError)({
                output,
                parsedBody,
                exceptionCtor: TimestreamWriteServiceException_1.TimestreamWriteServiceException,
                errorCode,
            });
    }
};
const deserializeAws_json1_0AccessDeniedExceptionResponse = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = deserializeAws_json1_0AccessDeniedException(body, context);
    const exception = new models_0_1.AccessDeniedException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const deserializeAws_json1_0ConflictExceptionResponse = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = deserializeAws_json1_0ConflictException(body, context);
    const exception = new models_0_1.ConflictException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const deserializeAws_json1_0InternalServerExceptionResponse = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = deserializeAws_json1_0InternalServerException(body, context);
    const exception = new models_0_1.InternalServerException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const deserializeAws_json1_0InvalidEndpointExceptionResponse = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = deserializeAws_json1_0InvalidEndpointException(body, context);
    const exception = new models_0_1.InvalidEndpointException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const deserializeAws_json1_0RejectedRecordsExceptionResponse = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = deserializeAws_json1_0RejectedRecordsException(body, context);
    const exception = new models_0_1.RejectedRecordsException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const deserializeAws_json1_0ResourceNotFoundExceptionResponse = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = deserializeAws_json1_0ResourceNotFoundException(body, context);
    const exception = new models_0_1.ResourceNotFoundException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const deserializeAws_json1_0ServiceQuotaExceededExceptionResponse = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = deserializeAws_json1_0ServiceQuotaExceededException(body, context);
    const exception = new models_0_1.ServiceQuotaExceededException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const deserializeAws_json1_0ThrottlingExceptionResponse = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = deserializeAws_json1_0ThrottlingException(body, context);
    const exception = new models_0_1.ThrottlingException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const deserializeAws_json1_0ValidationExceptionResponse = async (parsedOutput, context) => {
    const body = parsedOutput.body;
    const deserialized = deserializeAws_json1_0ValidationException(body, context);
    const exception = new models_0_1.ValidationException({
        $metadata: deserializeMetadata(parsedOutput),
        ...deserialized,
    });
    return (0, smithy_client_1.decorateServiceException)(exception, body);
};
const serializeAws_json1_0CreateDatabaseRequest = (input, context) => {
    return {
        ...(input.DatabaseName != null && { DatabaseName: input.DatabaseName }),
        ...(input.KmsKeyId != null && { KmsKeyId: input.KmsKeyId }),
        ...(input.Tags != null && { Tags: serializeAws_json1_0TagList(input.Tags, context) }),
    };
};
const serializeAws_json1_0CreateTableRequest = (input, context) => {
    return {
        ...(input.DatabaseName != null && { DatabaseName: input.DatabaseName }),
        ...(input.MagneticStoreWriteProperties != null && {
            MagneticStoreWriteProperties: serializeAws_json1_0MagneticStoreWriteProperties(input.MagneticStoreWriteProperties, context),
        }),
        ...(input.RetentionProperties != null && {
            RetentionProperties: serializeAws_json1_0RetentionProperties(input.RetentionProperties, context),
        }),
        ...(input.TableName != null && { TableName: input.TableName }),
        ...(input.Tags != null && { Tags: serializeAws_json1_0TagList(input.Tags, context) }),
    };
};
const serializeAws_json1_0DeleteDatabaseRequest = (input, context) => {
    return {
        ...(input.DatabaseName != null && { DatabaseName: input.DatabaseName }),
    };
};
const serializeAws_json1_0DeleteTableRequest = (input, context) => {
    return {
        ...(input.DatabaseName != null && { DatabaseName: input.DatabaseName }),
        ...(input.TableName != null && { TableName: input.TableName }),
    };
};
const serializeAws_json1_0DescribeDatabaseRequest = (input, context) => {
    return {
        ...(input.DatabaseName != null && { DatabaseName: input.DatabaseName }),
    };
};
const serializeAws_json1_0DescribeEndpointsRequest = (input, context) => {
    return {};
};
const serializeAws_json1_0DescribeTableRequest = (input, context) => {
    return {
        ...(input.DatabaseName != null && { DatabaseName: input.DatabaseName }),
        ...(input.TableName != null && { TableName: input.TableName }),
    };
};
const serializeAws_json1_0Dimension = (input, context) => {
    return {
        ...(input.DimensionValueType != null && { DimensionValueType: input.DimensionValueType }),
        ...(input.Name != null && { Name: input.Name }),
        ...(input.Value != null && { Value: input.Value }),
    };
};
const serializeAws_json1_0Dimensions = (input, context) => {
    return input
        .filter((e) => e != null)
        .map((entry) => {
        return serializeAws_json1_0Dimension(entry, context);
    });
};
const serializeAws_json1_0ListDatabasesRequest = (input, context) => {
    return {
        ...(input.MaxResults != null && { MaxResults: input.MaxResults }),
        ...(input.NextToken != null && { NextToken: input.NextToken }),
    };
};
const serializeAws_json1_0ListTablesRequest = (input, context) => {
    return {
        ...(input.DatabaseName != null && { DatabaseName: input.DatabaseName }),
        ...(input.MaxResults != null && { MaxResults: input.MaxResults }),
        ...(input.NextToken != null && { NextToken: input.NextToken }),
    };
};
const serializeAws_json1_0ListTagsForResourceRequest = (input, context) => {
    return {
        ...(input.ResourceARN != null && { ResourceARN: input.ResourceARN }),
    };
};
const serializeAws_json1_0MagneticStoreRejectedDataLocation = (input, context) => {
    return {
        ...(input.S3Configuration != null && {
            S3Configuration: serializeAws_json1_0S3Configuration(input.S3Configuration, context),
        }),
    };
};
const serializeAws_json1_0MagneticStoreWriteProperties = (input, context) => {
    return {
        ...(input.EnableMagneticStoreWrites != null && { EnableMagneticStoreWrites: input.EnableMagneticStoreWrites }),
        ...(input.MagneticStoreRejectedDataLocation != null && {
            MagneticStoreRejectedDataLocation: serializeAws_json1_0MagneticStoreRejectedDataLocation(input.MagneticStoreRejectedDataLocation, context),
        }),
    };
};
const serializeAws_json1_0MeasureValue = (input, context) => {
    return {
        ...(input.Name != null && { Name: input.Name }),
        ...(input.Type != null && { Type: input.Type }),
        ...(input.Value != null && { Value: input.Value }),
    };
};
const serializeAws_json1_0MeasureValues = (input, context) => {
    return input
        .filter((e) => e != null)
        .map((entry) => {
        return serializeAws_json1_0MeasureValue(entry, context);
    });
};
const serializeAws_json1_0_Record = (input, context) => {
    return {
        ...(input.Dimensions != null && { Dimensions: serializeAws_json1_0Dimensions(input.Dimensions, context) }),
        ...(input.MeasureName != null && { MeasureName: input.MeasureName }),
        ...(input.MeasureValue != null && { MeasureValue: input.MeasureValue }),
        ...(input.MeasureValueType != null && { MeasureValueType: input.MeasureValueType }),
        ...(input.MeasureValues != null && {
            MeasureValues: serializeAws_json1_0MeasureValues(input.MeasureValues, context),
        }),
        ...(input.Time != null && { Time: input.Time }),
        ...(input.TimeUnit != null && { TimeUnit: input.TimeUnit }),
        ...(input.Version != null && { Version: input.Version }),
    };
};
const serializeAws_json1_0Records = (input, context) => {
    return input
        .filter((e) => e != null)
        .map((entry) => {
        return serializeAws_json1_0_Record(entry, context);
    });
};
const serializeAws_json1_0RetentionProperties = (input, context) => {
    return {
        ...(input.MagneticStoreRetentionPeriodInDays != null && {
            MagneticStoreRetentionPeriodInDays: input.MagneticStoreRetentionPeriodInDays,
        }),
        ...(input.MemoryStoreRetentionPeriodInHours != null && {
            MemoryStoreRetentionPeriodInHours: input.MemoryStoreRetentionPeriodInHours,
        }),
    };
};
const serializeAws_json1_0S3Configuration = (input, context) => {
    return {
        ...(input.BucketName != null && { BucketName: input.BucketName }),
        ...(input.EncryptionOption != null && { EncryptionOption: input.EncryptionOption }),
        ...(input.KmsKeyId != null && { KmsKeyId: input.KmsKeyId }),
        ...(input.ObjectKeyPrefix != null && { ObjectKeyPrefix: input.ObjectKeyPrefix }),
    };
};
const serializeAws_json1_0Tag = (input, context) => {
    return {
        ...(input.Key != null && { Key: input.Key }),
        ...(input.Value != null && { Value: input.Value }),
    };
};
const serializeAws_json1_0TagKeyList = (input, context) => {
    return input
        .filter((e) => e != null)
        .map((entry) => {
        return entry;
    });
};
const serializeAws_json1_0TagList = (input, context) => {
    return input
        .filter((e) => e != null)
        .map((entry) => {
        return serializeAws_json1_0Tag(entry, context);
    });
};
const serializeAws_json1_0TagResourceRequest = (input, context) => {
    return {
        ...(input.ResourceARN != null && { ResourceARN: input.ResourceARN }),
        ...(input.Tags != null && { Tags: serializeAws_json1_0TagList(input.Tags, context) }),
    };
};
const serializeAws_json1_0UntagResourceRequest = (input, context) => {
    return {
        ...(input.ResourceARN != null && { ResourceARN: input.ResourceARN }),
        ...(input.TagKeys != null && { TagKeys: serializeAws_json1_0TagKeyList(input.TagKeys, context) }),
    };
};
const serializeAws_json1_0UpdateDatabaseRequest = (input, context) => {
    return {
        ...(input.DatabaseName != null && { DatabaseName: input.DatabaseName }),
        ...(input.KmsKeyId != null && { KmsKeyId: input.KmsKeyId }),
    };
};
const serializeAws_json1_0UpdateTableRequest = (input, context) => {
    return {
        ...(input.DatabaseName != null && { DatabaseName: input.DatabaseName }),
        ...(input.MagneticStoreWriteProperties != null && {
            MagneticStoreWriteProperties: serializeAws_json1_0MagneticStoreWriteProperties(input.MagneticStoreWriteProperties, context),
        }),
        ...(input.RetentionProperties != null && {
            RetentionProperties: serializeAws_json1_0RetentionProperties(input.RetentionProperties, context),
        }),
        ...(input.TableName != null && { TableName: input.TableName }),
    };
};
const serializeAws_json1_0WriteRecordsRequest = (input, context) => {
    return {
        ...(input.CommonAttributes != null && {
            CommonAttributes: serializeAws_json1_0_Record(input.CommonAttributes, context),
        }),
        ...(input.DatabaseName != null && { DatabaseName: input.DatabaseName }),
        ...(input.Records != null && { Records: serializeAws_json1_0Records(input.Records, context) }),
        ...(input.TableName != null && { TableName: input.TableName }),
    };
};
const deserializeAws_json1_0AccessDeniedException = (output, context) => {
    return {
        Message: (0, smithy_client_1.expectString)(output.Message),
    };
};
const deserializeAws_json1_0ConflictException = (output, context) => {
    return {
        Message: (0, smithy_client_1.expectString)(output.Message),
    };
};
const deserializeAws_json1_0CreateDatabaseResponse = (output, context) => {
    return {
        Database: output.Database != null ? deserializeAws_json1_0Database(output.Database, context) : undefined,
    };
};
const deserializeAws_json1_0CreateTableResponse = (output, context) => {
    return {
        Table: output.Table != null ? deserializeAws_json1_0Table(output.Table, context) : undefined,
    };
};
const deserializeAws_json1_0Database = (output, context) => {
    return {
        Arn: (0, smithy_client_1.expectString)(output.Arn),
        CreationTime: output.CreationTime != null
            ? (0, smithy_client_1.expectNonNull)((0, smithy_client_1.parseEpochTimestamp)((0, smithy_client_1.expectNumber)(output.CreationTime)))
            : undefined,
        DatabaseName: (0, smithy_client_1.expectString)(output.DatabaseName),
        KmsKeyId: (0, smithy_client_1.expectString)(output.KmsKeyId),
        LastUpdatedTime: output.LastUpdatedTime != null
            ? (0, smithy_client_1.expectNonNull)((0, smithy_client_1.parseEpochTimestamp)((0, smithy_client_1.expectNumber)(output.LastUpdatedTime)))
            : undefined,
        TableCount: (0, smithy_client_1.expectLong)(output.TableCount),
    };
};
const deserializeAws_json1_0DatabaseList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        if (entry === null) {
            return null;
        }
        return deserializeAws_json1_0Database(entry, context);
    });
    return retVal;
};
const deserializeAws_json1_0DescribeDatabaseResponse = (output, context) => {
    return {
        Database: output.Database != null ? deserializeAws_json1_0Database(output.Database, context) : undefined,
    };
};
const deserializeAws_json1_0DescribeEndpointsResponse = (output, context) => {
    return {
        Endpoints: output.Endpoints != null ? deserializeAws_json1_0Endpoints(output.Endpoints, context) : undefined,
    };
};
const deserializeAws_json1_0DescribeTableResponse = (output, context) => {
    return {
        Table: output.Table != null ? deserializeAws_json1_0Table(output.Table, context) : undefined,
    };
};
const deserializeAws_json1_0Endpoint = (output, context) => {
    return {
        Address: (0, smithy_client_1.expectString)(output.Address),
        CachePeriodInMinutes: (0, smithy_client_1.expectLong)(output.CachePeriodInMinutes),
    };
};
const deserializeAws_json1_0Endpoints = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        if (entry === null) {
            return null;
        }
        return deserializeAws_json1_0Endpoint(entry, context);
    });
    return retVal;
};
const deserializeAws_json1_0InternalServerException = (output, context) => {
    return {
        Message: (0, smithy_client_1.expectString)(output.Message),
    };
};
const deserializeAws_json1_0InvalidEndpointException = (output, context) => {
    return {
        Message: (0, smithy_client_1.expectString)(output.Message),
    };
};
const deserializeAws_json1_0ListDatabasesResponse = (output, context) => {
    return {
        Databases: output.Databases != null ? deserializeAws_json1_0DatabaseList(output.Databases, context) : undefined,
        NextToken: (0, smithy_client_1.expectString)(output.NextToken),
    };
};
const deserializeAws_json1_0ListTablesResponse = (output, context) => {
    return {
        NextToken: (0, smithy_client_1.expectString)(output.NextToken),
        Tables: output.Tables != null ? deserializeAws_json1_0TableList(output.Tables, context) : undefined,
    };
};
const deserializeAws_json1_0ListTagsForResourceResponse = (output, context) => {
    return {
        Tags: output.Tags != null ? deserializeAws_json1_0TagList(output.Tags, context) : undefined,
    };
};
const deserializeAws_json1_0MagneticStoreRejectedDataLocation = (output, context) => {
    return {
        S3Configuration: output.S3Configuration != null
            ? deserializeAws_json1_0S3Configuration(output.S3Configuration, context)
            : undefined,
    };
};
const deserializeAws_json1_0MagneticStoreWriteProperties = (output, context) => {
    return {
        EnableMagneticStoreWrites: (0, smithy_client_1.expectBoolean)(output.EnableMagneticStoreWrites),
        MagneticStoreRejectedDataLocation: output.MagneticStoreRejectedDataLocation != null
            ? deserializeAws_json1_0MagneticStoreRejectedDataLocation(output.MagneticStoreRejectedDataLocation, context)
            : undefined,
    };
};
const deserializeAws_json1_0RecordsIngested = (output, context) => {
    return {
        MagneticStore: (0, smithy_client_1.expectInt32)(output.MagneticStore),
        MemoryStore: (0, smithy_client_1.expectInt32)(output.MemoryStore),
        Total: (0, smithy_client_1.expectInt32)(output.Total),
    };
};
const deserializeAws_json1_0RejectedRecord = (output, context) => {
    return {
        ExistingVersion: (0, smithy_client_1.expectLong)(output.ExistingVersion),
        Reason: (0, smithy_client_1.expectString)(output.Reason),
        RecordIndex: (0, smithy_client_1.expectInt32)(output.RecordIndex),
    };
};
const deserializeAws_json1_0RejectedRecords = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        if (entry === null) {
            return null;
        }
        return deserializeAws_json1_0RejectedRecord(entry, context);
    });
    return retVal;
};
const deserializeAws_json1_0RejectedRecordsException = (output, context) => {
    return {
        Message: (0, smithy_client_1.expectString)(output.Message),
        RejectedRecords: output.RejectedRecords != null
            ? deserializeAws_json1_0RejectedRecords(output.RejectedRecords, context)
            : undefined,
    };
};
const deserializeAws_json1_0ResourceNotFoundException = (output, context) => {
    return {
        Message: (0, smithy_client_1.expectString)(output.Message),
    };
};
const deserializeAws_json1_0RetentionProperties = (output, context) => {
    return {
        MagneticStoreRetentionPeriodInDays: (0, smithy_client_1.expectLong)(output.MagneticStoreRetentionPeriodInDays),
        MemoryStoreRetentionPeriodInHours: (0, smithy_client_1.expectLong)(output.MemoryStoreRetentionPeriodInHours),
    };
};
const deserializeAws_json1_0S3Configuration = (output, context) => {
    return {
        BucketName: (0, smithy_client_1.expectString)(output.BucketName),
        EncryptionOption: (0, smithy_client_1.expectString)(output.EncryptionOption),
        KmsKeyId: (0, smithy_client_1.expectString)(output.KmsKeyId),
        ObjectKeyPrefix: (0, smithy_client_1.expectString)(output.ObjectKeyPrefix),
    };
};
const deserializeAws_json1_0ServiceQuotaExceededException = (output, context) => {
    return {
        Message: (0, smithy_client_1.expectString)(output.Message),
    };
};
const deserializeAws_json1_0Table = (output, context) => {
    return {
        Arn: (0, smithy_client_1.expectString)(output.Arn),
        CreationTime: output.CreationTime != null
            ? (0, smithy_client_1.expectNonNull)((0, smithy_client_1.parseEpochTimestamp)((0, smithy_client_1.expectNumber)(output.CreationTime)))
            : undefined,
        DatabaseName: (0, smithy_client_1.expectString)(output.DatabaseName),
        LastUpdatedTime: output.LastUpdatedTime != null
            ? (0, smithy_client_1.expectNonNull)((0, smithy_client_1.parseEpochTimestamp)((0, smithy_client_1.expectNumber)(output.LastUpdatedTime)))
            : undefined,
        MagneticStoreWriteProperties: output.MagneticStoreWriteProperties != null
            ? deserializeAws_json1_0MagneticStoreWriteProperties(output.MagneticStoreWriteProperties, context)
            : undefined,
        RetentionProperties: output.RetentionProperties != null
            ? deserializeAws_json1_0RetentionProperties(output.RetentionProperties, context)
            : undefined,
        TableName: (0, smithy_client_1.expectString)(output.TableName),
        TableStatus: (0, smithy_client_1.expectString)(output.TableStatus),
    };
};
const deserializeAws_json1_0TableList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        if (entry === null) {
            return null;
        }
        return deserializeAws_json1_0Table(entry, context);
    });
    return retVal;
};
const deserializeAws_json1_0Tag = (output, context) => {
    return {
        Key: (0, smithy_client_1.expectString)(output.Key),
        Value: (0, smithy_client_1.expectString)(output.Value),
    };
};
const deserializeAws_json1_0TagList = (output, context) => {
    const retVal = (output || [])
        .filter((e) => e != null)
        .map((entry) => {
        if (entry === null) {
            return null;
        }
        return deserializeAws_json1_0Tag(entry, context);
    });
    return retVal;
};
const deserializeAws_json1_0TagResourceResponse = (output, context) => {
    return {};
};
const deserializeAws_json1_0ThrottlingException = (output, context) => {
    return {
        Message: (0, smithy_client_1.expectString)(output.Message),
    };
};
const deserializeAws_json1_0UntagResourceResponse = (output, context) => {
    return {};
};
const deserializeAws_json1_0UpdateDatabaseResponse = (output, context) => {
    return {
        Database: output.Database != null ? deserializeAws_json1_0Database(output.Database, context) : undefined,
    };
};
const deserializeAws_json1_0UpdateTableResponse = (output, context) => {
    return {
        Table: output.Table != null ? deserializeAws_json1_0Table(output.Table, context) : undefined,
    };
};
const deserializeAws_json1_0ValidationException = (output, context) => {
    return {
        Message: (0, smithy_client_1.expectString)(output.Message),
    };
};
const deserializeAws_json1_0WriteRecordsResponse = (output, context) => {
    return {
        RecordsIngested: output.RecordsIngested != null
            ? deserializeAws_json1_0RecordsIngested(output.RecordsIngested, context)
            : undefined,
    };
};
const deserializeMetadata = (output) => ({
    httpStatusCode: output.statusCode,
    requestId: output.headers["x-amzn-requestid"] ?? output.headers["x-amzn-request-id"] ?? output.headers["x-amz-request-id"],
    extendedRequestId: output.headers["x-amz-id-2"],
    cfId: output.headers["x-amz-cf-id"],
});
const collectBody = (streamBody = new Uint8Array(), context) => {
    if (streamBody instanceof Uint8Array) {
        return Promise.resolve(streamBody);
    }
    return context.streamCollector(streamBody) || Promise.resolve(new Uint8Array());
};
const collectBodyString = (streamBody, context) => collectBody(streamBody, context).then((body) => context.utf8Encoder(body));
const buildHttpRpcRequest = async (context, headers, path, resolvedHostname, body) => {
    const { hostname, protocol = "https", port, path: basePath } = await context.endpoint();
    const contents = {
        protocol,
        hostname,
        port,
        method: "POST",
        path: basePath.endsWith("/") ? basePath.slice(0, -1) + path : basePath + path,
        headers,
    };
    if (resolvedHostname !== undefined) {
        contents.hostname = resolvedHostname;
    }
    if (body !== undefined) {
        contents.body = body;
    }
    return new protocol_http_1.HttpRequest(contents);
};
const parseBody = (streamBody, context) => collectBodyString(streamBody, context).then((encoded) => {
    if (encoded.length) {
        return JSON.parse(encoded);
    }
    return {};
});
const parseErrorBody = async (errorBody, context) => {
    const value = await parseBody(errorBody, context);
    value.message = value.message ?? value.Message;
    return value;
};
const loadRestJsonErrorCode = (output, data) => {
    const findKey = (object, key) => Object.keys(object).find((k) => k.toLowerCase() === key.toLowerCase());
    const sanitizeErrorCode = (rawValue) => {
        let cleanValue = rawValue;
        if (typeof cleanValue === "number") {
            cleanValue = cleanValue.toString();
        }
        if (cleanValue.indexOf(",") >= 0) {
            cleanValue = cleanValue.split(",")[0];
        }
        if (cleanValue.indexOf(":") >= 0) {
            cleanValue = cleanValue.split(":")[0];
        }
        if (cleanValue.indexOf("#") >= 0) {
            cleanValue = cleanValue.split("#")[1];
        }
        return cleanValue;
    };
    const headerKey = findKey(output.headers, "x-amzn-errortype");
    if (headerKey !== undefined) {
        return sanitizeErrorCode(output.headers[headerKey]);
    }
    if (data.code !== undefined) {
        return sanitizeErrorCode(data.code);
    }
    if (data["__type"] !== undefined) {
        return sanitizeErrorCode(data["__type"]);
    }
};
